src/util/decoders.js

```diff
@@ -47,239 +51,266 @@ import { NETWORK_LIST, SUBSTRATE_NETWORK_LIST, SubstrateNetworkKeys } from '../c
-
-  const length8 = parseInt(rawData.substr(0, 2), 16) || 0;
-  const length16 = parseInt(rawData.substr(0, 4), 16) || 0;
-  let length = 0;
-
-  // Strip length prefix
-  if (length8 * 2 + 2 === rawData.length) {
-    rawData = rawData.substr(2);
-    length = length8;
-  } else if (length16 * 2 + 4 === rawData.length) {
-    rawData = rawData.substr(4);
-    length = length16;
-  } else {
-    return null;
-  }
-
-  const bytes = new Uint8Array(length);
-
-  for (let i = 0; i < length; i++) {
-    bytes[i] = parseInt(rawData.substr(i * 2, 2), 16);
-  }
-
-  return bytes;
+	if (!rawData) {
+		return null;
+	}
+
+	// Strip filler bytes padding at the end
+	if (rawData.substr(-2) === 'ec') {
+		rawData = rawData.substr(0, rawData.length - 2);
+	}
+
+	while (rawData.substr(-4) === 'ec11') {
+		rawData = rawData.substr(0, rawData.length - 4);
+	}
+
+	// Verify that the QR encoding is binary and it's ending with a proper terminator
+	if (rawData.substr(0, 1) !== '4' || rawData.substr(-1) !== '0') {
+		return null;
+	}
+
+	// Strip the encoding indicator and terminator for ease of reading
+	rawData = rawData.substr(1, rawData.length - 2);
+
+	const length8 = parseInt(rawData.substr(0, 2), 16) || 0;
+	const length16 = parseInt(rawData.substr(0, 4), 16) || 0;
+	let length = 0;
+
+	// Strip length prefix
+	if (length8 * 2 + 2 === rawData.length) {
+		rawData = rawData.substr(2);
+		length = length8;
+	} else if (length16 * 2 + 4 === rawData.length) {
+		rawData = rawData.substr(4);
+		length = length16;
+	} else {
+		return null;
+	}
+
+	const bytes = new Uint8Array(length);
+
+	for (let i = 0; i < length; i++) {
+		bytes[i] = parseInt(rawData.substr(i * 2, 2), 16);
+	}
+
+	return bytes;
 }
 
 export async function constructDataFromBytes(bytes) {
-  const frameInfo = hexStripPrefix(u8aToHex(bytes.slice(0, 5)));
-  const isMultipart = !!parseInt(frameInfo.substr(0, 2), 16);
-  const frameCount = parseInt(frameInfo.substr(2, 4), 16);
-  const currentFrame = parseInt(frameInfo.substr(6, 4), 16);
-  const uosAfterFrames = hexStripPrefix(u8aToHex(bytes.slice(5)));
-
-  if (isMultipart) {
-    const partData = {
-      currentFrame,
-      frameCount,
-      isMultipart: true,
-      partData: uosAfterFrames
-    };
-
-    return partData;
-  }
-
-  const zerothByte = uosAfterFrames.substr(0, 2);
-  const firstByte = uosAfterFrames.substr(2, 2);
-  const secondByte = uosAfterFrames.substr(4, 2);
-  let action;
-  let address;
-  let data = {};
-  data['data'] = {}; // for consistency with legacy data format.
-
-  try {
-    // decode payload appropriately via UOS
-    switch (zerothByte) {
-      case '45': // Ethereum UOS payload
-        action =
-          firstByte === '00' || firstByte === '01'
-            ? 'signData'
-            : firstByte === '01'
-            ? 'signTransaction'
-            : null;
-        address = uosAfterFrames.substr(4, 44);
-
-        data['action'] = action;
-        data['data']['account'] = account;
```

Comment:
The account value should be assigned to `address` rather than `account`

Reply:
Not sure I understand the comment.
Do you mean we should rename it? If yes then let's do it in another PR.

Reply:
This `account` variable is undefined in the file scope, I think it was a mistake, it should be `address` variable.

Reply:
thanks for catching that. exactly why we need a linter. IMO this is fine to merge now, as only Ethereum UOS will be affected (which is not supported by anyone at the moment).